dll/win32/shell32/shellrecyclebin/recyclebin_v5_enumerator.c

```diff
@@ -210,6 +210,43 @@ RecycleBin5File_RecycleBinFile_GetFileName(
     return S_OK;
 }
 
+static HRESULT STDMETHODCALLTYPE
+RecycleBin5File_RecycleBinFile_GetTypeName(
+    IN IRecycleBinFile *This,
+    IN SIZE_T BufferSize,
+    IN OUT LPWSTR Buffer,
+    OUT SIZE_T *RequiredSize)
+{
+    HRESULT hr;
+
+    struct RecycleBin5File *s = CONTAINING_RECORD(This, struct RecycleBin5File, recycleBinFileImpl);
+    DWORD dwRequired;
+    DWORD dwAttributes;
+    SHFILEINFOW shFileInfo;
+
+    TRACE("(%p, %u, %p, %p)\n", This, BufferSize, Buffer, RequiredSize);
+
+    hr = RecycleBin5File_RecycleBinFile_GetAttributes(This, &dwAttributes);
+    if (!SUCCEEDED(hr))
+        return hr;
+
+    hr = SHGetFileInfoW(s->FullName, dwAttributes, &shFileInfo, sizeof(shFileInfo), SHGFI_TYPENAME | SHGFI_USEFILEATTRIBUTES);
+    if (!SUCCEEDED(hr))
+        return hr;
+
+    dwRequired = (DWORD)(wcslen(shFileInfo.szTypeName) + 1) * sizeof(WCHAR);
+    if (RequiredSize)
+        *RequiredSize = dwRequired;
+
+    if (BufferSize == 0 && !Buffer)
```

Comment:
What if someone passes a NULL `Buffer` but a non-zero `BufferSize` ?

Reply:
(OK I see it's copy-pasted from already-existing `RecycleBin5File_RecycleBinFile_GetFileName` ...)

Reply:
Really idk, the original code was like that, look at the function above this function